### PR TITLE
Split up notebook check

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Interim test for https://github.com/Infleqtion/client-superstaq/issues/889

(Also re-runs some notebooks for updated target outputs and adds the `available=True` flag in `_HPCA_Tutorial.ipynb`)

### CODE REVIEW

**Comment by bharat-thotakura** [comment]
Reminder: https://github.com/Infleqtion/client-superstaq/issues/752

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -238,11 +238,23 @@ jobs:
...
-      SUPERSTAQ_API_KEY : ${{ secrets.SUPERSTAQ_API_KEY }}
-      IBMQ_TOKEN : ${{ secrets.IBMQ_TOKEN }}
+      SUPERSTAQ_API_KEY: ${{ secrets.SUPERSTAQ_API_KEY }}
+      IBMQ_TOKEN: ${{ secrets.IBMQ_TOKEN }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - type: "qiskit-superstaq"
+            pattern: "*_qss.ipynb"
```
Comment (richrines1) [inline .github/workflows/ci.yml:249]: notebooks from qiskit-superstaq/ should probably be included here as well
```suggestion
            pattern: "*_qss.ipynb qiskit-superstaq"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -238,11 +238,23 @@ jobs:
...
+            dependencies: "-e ./checks-superstaq -e ./general-superstaq[dev] -e ./qiskit-superstaq[dev]"
+          - type: "cirq-superstaq"
+            pattern: "*_css.ipynb"
```
Comment (richrines1) [inline .github/workflows/ci.yml:252]: ditto
```suggestion
            pattern: "*_css.ipynb cirq-superstaq"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -238,11 +238,23 @@ jobs:
...
+            dependencies: "-e ./checks-superstaq -e ./general-superstaq[dev] -e ./cirq-superstaq[dev]"
+          - type: "other docs"
+            pattern: "--exclude=**/*_css.ipynb --exclude=**/*_qss.ipynb"
```
Comment (richrines1) [inline .github/workflows/ci.yml:255]: assuming prior (+ cleanup)
```suggestion
            pattern: "--exclude *_css.ipynb *_qss.ipynb 'cirq-superstaq/*' 'qiskit-superstaq/*'"
```

i think all this leaves is `ibmq_dd_strategies` and the HPCA tutorial though. The former could probably just be renamed `ibmq_dd_strategies_qss` (it doesn't use css at all), and the latter should maybe be split up into qss and css versions? (definitely outside the scope of this pr though)

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -238,11 +238,23 @@ jobs:
...
-      IBMQ_TOKEN : ${{ secrets.IBMQ_TOKEN }}
+      SUPERSTAQ_API_KEY: ${{ secrets.SUPERSTAQ_API_KEY }}
+      IBMQ_TOKEN: ${{ secrets.IBMQ_TOKEN }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - type: "qiskit-superstaq"
+            pattern: "*_qss.ipynb"
+            dependencies: "-e ./checks-superstaq -e ./general-superstaq[dev] -e ./qiskit-superstaq[dev]"
```
Comment (richrines1) [inline .github/workflows/ci.yml:250]: imo we should also remove the actions/cache... block from these tests and change this to
```suggestion
            dependencies: "-e ./checks-superstaq -e ./general-superstaq -e ./qiskit-superstaq[examples]"
```
so this doubles as a test of our example requirements (and hopefully catches things like #885)

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -238,11 +238,23 @@ jobs:
...
+          - type: "cirq-superstaq"
+            pattern: "*_css.ipynb"
+            dependencies: "-e ./checks-superstaq -e ./general-superstaq[dev] -e ./cirq-superstaq[dev]"
```
Comment (richrines1) [inline .github/workflows/ci.yml:253]:
```suggestion
            dependencies: "-e ./checks-superstaq -e ./general-superstaq -e ./cirq-superstaq[examples]"
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -238,28 +238,36 @@ jobs:
...
+        include:
+          - type: "qiskit-superstaq"
+            pattern: "*_qss.ipynb qiskit-superstaq"
+            dependencies: "-e ./checks-superstaq -e ./general-superstaq -e ./qiskit-superstaq[examples]"
+          - type: "cirq-superstaq"
+            pattern: "*_css.ipynb cirq-superstaq"
+            dependencies: "-e ./checks-superstaq -e ./general-superstaq -e ./cirq-superstaq[examples]"
+          - type: "other docs"
+            pattern: "--exclude *_css.ipynb *_qss.ipynb 'cirq-superstaq/*' 'qiskit-superstaq/*'"
+            dependencies: "-e ./checks-superstaq -e ./general-superstaq[dev] -e ./qiskit-superstaq[dev] -e ./cirq-superstaq[dev] -e ./supermarq-benchmarks[dev]"
```
Comment (richrines1) [inline .github/workflows/ci.yml:256]: (sorry one more nit)
```suggestion
            dependencies: "-e ./checks-superstaq -e ./general-superstaq -e ./qiskit-superstaq[examples] -e ./cirq-superstaq[examples] -e ./supermarq-benchmarks"
```

**Review by richrines1 — APPROVED** [review]
this lgtm! when you're ready to merge we can switch the github rules to the new set of checks

thanks for updating this :)

**Comment by bharat-thotakura** [comment]
> this lgtm! when you're ready to merge we can switch the github rules to the new set of checks
> 
> thanks for updating this :)

Thanks! I'm good to merge this but also opened https://github.com/Infleqtion/client-superstaq/pull/893/files as a brief solution to drop the 'Notebook check for other docs' now rather than later -- if https://github.com/Infleqtion/client-superstaq/pull/893 looks good. Otherwise, we can keep the current update as is.

**Comment by bharat-thotakura** [comment]
Update: With https://github.com/Infleqtion/client-superstaq/pull/893 merged, this PR has been updated to only have the 'Notebook check for qiskit/cirq-superstaq' 

cc: @vtomole @richrines1

**Comment by richrines1** [comment]
updated the requirements, so should be good to merge!

given that the checks still takes a while there might be an argument for splitting "supermarq" off into its own category down the line...

**Comment by bharat-thotakura** [comment]
^ I'm hoping the resolution to https://github.com/Infleqtion/client-superstaq/issues/772 would void the need for a further update since 'Notebook check for cirq-superstaq' finishes much faster and the cause seems to be the call to `.backends()`

**Comment by richrines1** [comment]
> ^ I'm hoping the resolution to #772 would void the need for a further update since 'Notebook check for cirq-superstaq' finishes much faster and the cause seems to be the call to `.backends()`

ahh yep that makes sense

**Comment by bharat-thotakura** [comment]
Made some interim changes with [1f953b0](https://github.com/Infleqtion/client-superstaq/pull/892/commits/1f953b0ca01fc19b907e4a889d846beaf9333e0e) that should hopefully cut down the `qss` notebook check time.

File: supermarq-benchmarks/examples/Supermarq_HPCA_Tutorial_css.ipynb
Comment (richrines1) [inline supermarq-benchmarks/examples/Supermarq_HPCA_Tutorial_css.ipynb:1]: do you know why this file has such a large diff?

Comment (bharat-thotakura) [reply supermarq-benchmarks/examples/Supermarq_HPCA_Tutorial_css.ipynb:1]: In trying to resolve the merge conflict, I manually resolved the changes and pushed ... and that seems to have caused. Looking at the notebook differences though, nothing new should have been added (except for a markdown cell update: 'Service / Provider' -> 'Service')

**Pull request merged**